tmt/base.py

```diff
@@ -739,7 +739,17 @@ def __init__(self, id_=None, tree=None, context=None):
         self.config.last_run(self.workdir)
         # Save the tree
         try:
-            self.tree = tree if tree else tmt.Tree('.')
+            # If run id was given and root was not explicitly specified,
+            # create a new Tree from the root in run.yaml
+            if id_ and context.parent.params and not context.parent.params['root']:
+                try:
+                    data = tmt.utils.yaml_to_dict(self.read('run.yaml'))
```

Comment:
Let's move this into the [load](https://github.com/psss/tmt/blob/master/tmt/base.py#L776) method so that we do not read the `run.yaml` file twice.